### PR TITLE
Enable HA mode for Sentry

### LINKED ISSUE #2180: Daprd sidecar warnings on loading components in k8s 1.18.8
Seeing a bunch of these warnings in Dapr `v.0.11` sidecar logs that seem to be related to the new SPIFFE identity support. I haven't had a chance to check on any other version of k8s besides `v1.18.8`

```json
{
  "app_id": "reduced",
  "instance": "reduced",
  "level": "warning",
  "msg": "failed to load components: rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority (possibly because of \\\"x509: ECDSA verification failure\\\" while trying to verify candidate authority certificate \\\"cluster.local\\\")\"",
  "scope": "dapr.runtime",
  "time": "2020-10-02T17:46:31.026244212Z",
  "type": "log",
  "ver": "0.11.0"
}
```

So far, not seeing any issues with the specific components but I've only tested like 5-7 of them. 

> /area runtime

### LINKED ISSUE #2187: HA mode is broken
First reported by @mchmarny [here](https://github.com/dapr/dapr/issues/2180).

In 0.11.0, enabling HA mode will cause sidecars to report authentication issues with the Dapr control plane.

This happens because multiple instances of Sentry will boot up and create their own root certs, while only one instance will save the certs for future use in a secret or on the filesystem.
This means that until the Sentry pods are restarted, there's a situation where N root certs exist in a cluster.

### PR DESCRIPTION
**Pull request by yaron2**
Closes #2187 

This PR adds an additional check for Sentry to see if credentials are already created, and if so wait for them to be detected on the filesystem before loading them.

In addition, this PR removes a redundant channel.

This fixes a (*possible) regression in 0.11.0 where multiple sentry pods would create the certs at the same time and until the pods were restarted, the cluster would exist with 3 different root CAs loaded in memory.

* possible because, although weird, this wasn't reproduced in 0.10.0

### CODE REVIEW

**Comment by youngbupark** [comment]
Please add tests for this change.

**Comment by youngbupark** [comment]
enabling HA on e2e test would not be able to 100% catch this bug. I would add unit-test to validate `detectCertificates` and `shouldCreateCerts` to prevent the regression.

**Comment by yaron2** [comment]
> enabling HA on e2e test would not be able to 100% catch this bug. I would add unit-test to validate `detectCertificates` and `shouldCreateCerts` to prevent the regression.

Writing the tests for these two cases now.

However, HA on e2e is actually what I did to catch this bug.

**Comment by youngbupark** [comment]
> > enabling HA on e2e test would not be able to 100% catch this bug. I would add unit-test to validate `detectCertificates` and `shouldCreateCerts` to prevent the regression.
> 
> Writing the tests for these two cases now.

Thanks.

> 
> However, HA on e2e is actually what I did to catch this bug.

I see. I agree on having HA mode on e2e test. The targeted unit-test will help a lot to find the issue earlier.

File: pkg/sentry/ca/certificate_authority.go [hunk 1]
```diff
@@ -137,13 +146,36 @@ func shouldCreateCerts(conf config.SentryConfig) bool {
 	return len(b) == 0
 }
 
+func detectCertificates(path string) error {
+	t := time.NewTicker(time.Second * 1)
```
Comment (youngbupark) [inline pkg/sentry/ca/certificate_authority.go:150]: Can you please define `time.Second * 1` as const?

**Review by youngbupark — CHANGES_REQUESTED** [review]
Thanks for adding tests. I left one more small comment.

File: pkg/sentry/ca/certificate_authority.go (hunk 1 above)
Comment (yaron2) [reply pkg/sentry/ca/certificate_authority.go]: Done

**Review by youngbupark — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**